### ISSUE TITLE
gas estimation fails on expensive contract functions

### ISSUE THREAD

**Issue by soundreactor** [INFURA/infura#108]
im using Infura with web3.py to estimate gas  for contract function transactions.
at first i thought its was a web3 problem: https://github.com/ethereum/web3.py/issues/845

this works fine for functions that require ~50000 gas

however Infura seems block estimation for functions that require more gas for example ~300000

can i increase the limit? is this desired behavior ?

**Comment by tueric** [comment]
No, this is not desired behaviour nor is it something we've encountered yet... It would be useful if you could give us the request your are trying to make. The error message would be awesome too.

**Comment by peterchealse** [comment]
long time passed, i have the same problem.. any solution?

**Comment by Yakelixir** [comment]
@soundreactor How did you resolve this?
@ec2 any results?

**Comment by soundreactor** [comment]
for me the contract function always used the same amount of gas so i just set it manually with some margin. i guess depends on your function. you might be fine.
